cdap-data-fabric-tests/src/test/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseMetricsTableTest.java

```diff
@@ -147,11 +153,97 @@ public void testConcurrentIncrement() throws Exception {
+    colIterator = secondRow.getColumns().entrySet().iterator();
+    column = colIterator.next();
+    Assert.assertEquals(1L, Bytes.toLong(column.getKey()));
+    Assert.assertEquals(1L, Bytes.toLong(column.getValue()));
+    column = colIterator.next();
+    Assert.assertEquals(2L, Bytes.toLong(column.getKey()));
+    Assert.assertEquals(4L, Bytes.toLong(column.getValue()));
+    column = colIterator.next();
+    Assert.assertEquals(3L, Bytes.toLong(column.getKey()));
+    Assert.assertEquals(3L, Bytes.toLong(column.getValue()));
+
+    Row thirdRow = combinedScanner.next();
+    Assert.assertEquals(3L, Bytes.toLong(thirdRow.getRow()));
+    colIterator = thirdRow.getColumns().entrySet().iterator();
+    column = colIterator.next();
+    Assert.assertEquals(4L, Bytes.toLong(column.getKey()));
+    Assert.assertEquals(4L, Bytes.toLong(column.getValue()));
+    column = colIterator.next();
+    Assert.assertEquals(7L, Bytes.toLong(column.getKey()));
+    Assert.assertEquals(7L, Bytes.toLong(column.getValue()));
+  }
+
+  @Test
+  public void testCombinedTablePut() throws Exception {
+    MetricsTable v2Table = getTable("v2Table");
+    MetricsTable v3Table = getTable("v3Table");
+
+    MetricsTable combinedMetricsTable = new CombinedHBaseMetricsTable(v2Table, v3Table);
+
+    combinedMetricsTable.put(ImmutableSortedMap.<byte[], SortedMap<byte[], Long>>orderedBy(Bytes.BYTES_COMPARATOR)
+                  .put(A, mapOf(A, Bytes.toLong(A), B, Bytes.toLong(B)))
+                  .put(B, mapOf(A, Bytes.toLong(A), B, Bytes.toLong(B))).build());
+    Assert.assertEquals(Bytes.toLong(A), Bytes.toLong(combinedMetricsTable.get(A, A)));
+    Assert.assertEquals(Bytes.toLong(B), Bytes.toLong(combinedMetricsTable.get(A, B)));
+    Assert.assertEquals(Bytes.toLong(A), Bytes.toLong(combinedMetricsTable.get(B, A)));
+    Assert.assertEquals(Bytes.toLong(B), Bytes.toLong(combinedMetricsTable.get(B, B)));
+
+    combinedMetricsTable.put(ImmutableSortedMap.<byte[], SortedMap<byte[], Long>>orderedBy(Bytes.BYTES_COMPARATOR)
+                  .put(B, mapOf(B, Bytes.toLong(B), C, Bytes.toLong(C)))
+                  .put(C, mapOf(P, Bytes.toLong(P), X, Bytes.toLong(X))).build());
+    Assert.assertEquals(Bytes.toLong(B), Bytes.toLong(combinedMetricsTable.get(B, B)));
+    Assert.assertEquals(Bytes.toLong(C), Bytes.toLong(combinedMetricsTable.get(B, C)));
+    Assert.assertEquals(Bytes.toLong(X), Bytes.toLong(combinedMetricsTable.get(C, X)));
+    Assert.assertEquals(Bytes.toLong(P), Bytes.toLong(combinedMetricsTable.get(C, P)));
+  }
+
   @Override
   protected MetricsTable getTable(String name) throws Exception {
     DatasetId metricsDatasetInstanceId = NamespaceId.SYSTEM.dataset(name);
-    DatasetProperties props = TableProperties.builder().setReadlessIncrementSupport(true).build();
+    DatasetProperties props = TableProperties.builder().setReadlessIncrementSupport(true)
+      .add(Constants.Metrics.METICS_HBASE_TABLE_SPLITS, 16).build();
```

Comment:
I realize now that this is in a test case. The test case should create the table in the same way that the metrics system would. Ideally call that method from the test case.